packages/@aws-cdk/assets/test/test.staging.ts

```diff
@@ -57,5 +57,22 @@ export = {
       'tree.json',
     ]);
     test.done();
+  },
+
+  'allow specifying extra data to include in the source hash'(test: Test) {
+    // GIVEN
+    const app = new App();
+    const stack = new Stack(app, 'stack');
+    const directory = path.join(__dirname, 'fs', 'fixtures', 'test1');
+
+    // WHEN
+    const withoutExtra = new Staging(stack, 'withoutExtra', { sourcePath: directory });
+    const withExtra = new Staging(stack, 'withExtra', { sourcePath: directory, extra: 'boom' });
```

Comment:
The `extra` here doesn't appear related to the fingerprint at all (the indication is entirely contained in the interface that declares the field initially). This doesn't read so friendly on the usage site :/

Reply:
Yeah, good point. Let me see if I can change this without backwards compat breakage. I suspect I can since `FingerprintOptions` was internal.

Reply:
Renamed to `extraHash`

Reply:
Maybe `hashSalt` or `fingerprintSalt` would be better (aka somewhat more descriptive), but `extraHash` is okay, too.